src/coord/src/catalog.rs

```diff
@@ -191,6 +191,16 @@ impl CatalogItem {
         }
     }
 
+    pub fn desc(&self) -> Option<&RelationDesc> {
+        match self {
+            CatalogItem::Table(t) => Some(&t.desc),
+            CatalogItem::Source(s) => Some(&s.desc),
+            CatalogItem::View(v) => Some(&v.desc),
+            CatalogItem::Sink(_) => None,
+            CatalogItem::Index(_) => None,
+        }
+    }
```

Comment:
A slightly different take on this method already exists as `CatalogEntry::desc`. Can you move that definition here (basically it returns a `Result` instead of an `Option`), and then adjust `CatalogEntry::desc` to simply delegate to `self.item.desc()`?